pkg/instance/instance.go

```diff
@@ -525,3 +535,26 @@ func RunnerCmd(prog, fwdAddr, os, arch string, poolIdx, vmIdx int, threaded, new
 	return fmt.Sprintf("%s -addr=%s -os=%s -arch=%s -pool=%d -vm=%d "+
 		"-threaded=%t -new-env=%t", prog, fwdAddr, os, arch, poolIdx, vmIdx, threaded, newEnv)
 }
+
+type BuildSemaphore chan struct{}
```

Comment:
It makes sense to make it a struct and pass by pointer explicitly.
When it was just a `chan` it was explicit that this is a reference type that should be passed by value. But which an opaque `BuildSemaphore` it's unclear and not extensible (e.g. if we will need to add a new field).